src/main/java/sirius/db/jdbc/BaseSQLQuery.java

```diff
@@ -58,14 +58,33 @@ public List<Row> queryList() throws SQLException {
     @Nonnull
     public List<Row> queryList(Limit limit) throws SQLException {
         List<Row> result = new ArrayList<>();
-        iterate(result::add, limit);
+        doIterate(result::add, limit);
 
         return result;
     }
 
+    /**
+     * Executes the given query by invoking the given <tt>handler</tt> for each result row as long as the current
+     * {@linkplain TaskContext task context} is active.
+     * <p>
+     * Consider using the method instead of {@link #queryList()} if a large result set is expected as this method. As
+     * this method only processes one row at a time, this might be much more memory efficient.
+     *
+     * @param handler the row handler invoked for each row
+     * @param limit   the limit which controls which and how many rows are output. Can be <tt>null</tt> to indicate
+     *                that there is no limit.
+     * @throws SQLException in case of a database error
+     */
+    public void iterate(Predicate<Row> handler, @Nullable Limit limit) throws SQLException {
+        doIterate(handler, limit);
```

Comment:
shouldn't the TaskContext be checked here?

Reply:
🙈